utils/scripts/ocm/ocm.py

```diff
@@ -1166,6 +1169,54 @@ def compare_with_old_version_file(self):
         log.info("File is updated to : {} ".format(old_data))
 
 
+    def change_cluster_channel_group(self):
+        """update the channel using ocm cmd"""
+        cluster_id = self.get_osd_cluster_id()
+        run_change_channel_cmd = (
+            "ocm patch /api/clusters_mgmt/v1/clusters/{}"
+            " --body {}".format(cluster_id, self.upadte_ocm_channel_json))
+        log.info(run_change_channel_cmd)
+        ret = execute_command(run_change_channel_cmd)
+        if ret is None:
+            log.info("Failed  to Update the channel")
+            return ret
+
+
+    def update_ocm_policy(self):
+        """upadte cluster policy to schedule for upgrade osd"""
+        cluster_id = self.get_osd_cluster_id()
+        utc_time_cmd = (
+            ''' oc debug node/"$(oc get nodes | awk 'FNR == 2 {print $1}')"\
+         -- chroot /host date -d '+7 min' -u '+%Y-%m-%dT%H:%M:%SZ' ''')
+
+        utc_time = execute_command(utc_time_cmd)
+        data = read_data_from_json(self.update_policies_json)
+        data["next_run"] = utc_time.replace("\n", "")
+
+        if data["version"] == "latest":
+            get_latest_upgradte_version = (
+                "ocm get cluster {} |"
+                " jq -r '.version.available_upgrades | values'".format(
+                    cluster_id))
+            latest_upgrade_version = execute_command(
+                get_latest_upgradte_version)
+            log.info("Version Available to Upgrade are ...{}".format(
+                latest_upgrade_version))
+            latest_upgrade_version = (
+                ast.literal_eval(latest_upgrade_version)[-1])
+
+            data["version"] = latest_upgrade_version
+
+        write_data_in_json(self.update_policies_json, data)
+        schedule_cluster_upgrade = (
+            "ocm post /api/clusters_mgmt/v1/clusters/{}/upgrade_policies"
+            " --body {}".format(cluster_id, self.update_policies_json))
+        ret = execute_command(schedule_cluster_upgrade)
+        if ret is None:
+            log.info("Failed  to Update the Upgrade Policy")
+            return ret
```

Comment:
Do we have to add a validation checking that the cluster upgrade that is scheduled is correct ?

Reply:
the tie we are taking  is from the node of openshift cluster so I guess we don't need it
the give time is in a wrong format or <5 min the ocm patch will fail